bin/journey-test.sh

```diff
@@ -233,15 +233,15 @@ solve_all_exercises() {
 
     pushd ${exercism_exercises_dir}/${TRACK}/${exercise}
     # Check that tests compile before we strip @Ignore annotations
-    "$EXECPATH"/gradlew compileTestJava
+    "$EXECPATH"/gradlew compileTestJava --configure-on-demand --parallel
```

Comment:
My hunch is that since tasks are being run for every subproject, then every subproject has to be configured _eventually_ within each build. That + natural variance could explain why in your second test runs, the configuration phase(s) of the version _with_ configuration on demand take longer in total!

My vote would be to remove this flag everywhere since the balance of extra complexity vs performance improvement seems unfavorable.